libs/ledger/src/lib/ledger-entries-data-provider.ts

```diff
@@ -51,28 +110,119 @@ const ledgerEntriesOnlyProvider = makeDataProvider<
 });
 
 export const ledgerEntriesProvider = makeDerivedDataProvider<
-  LedgerEntry[],
-  never
+  (AggregatedLedgerEntriesEdge | null)[],
+  AggregatedLedgerEntriesEdge[],
+  LedgerEntriesQueryVariables
 >(
   [ledgerEntriesOnlyProvider, assetsProvider, marketsProvider],
-  ([entries, assets, markets]): LedgerEntry[] =>
-    entries.map((entry: LedgerEntry) => {
+  ([entries, assets, markets]) => {
+    return entries.map((edge: AggregatedLedgerEntriesEdge) => {
+      const entry = edge?.node;
       const asset = assets.find((asset: Asset) => asset.id === entry.assetId);
       const marketSender = markets.find(
         (market: Market) => market.id === entry.senderMarketId
       );
       const marketReceiver = markets.find(
         (market: Market) => market.id === entry.receiverMarketId
       );
-      return { ...entry, asset, marketSender, marketReceiver };
-    })
+      return { node: { ...entry, asset, marketSender, marketReceiver } };
+    });
+  }
 );
 
-export const useLedgerEntriesDataProvider = (partyId: string) => {
-  const variables = useMemo(() => ({ partyId }), [partyId]);
-  return useDataProvider({
+interface Props {
+  partyId: string;
+  filter?: Filter;
+  gridRef: RefObject<AgGridReact>;
+  scrolledToTop: RefObject<boolean>;
+}
+
+export const useLedgerEntriesDataProvider = ({
+  partyId,
+  filter,
+  gridRef,
+  scrolledToTop,
+}: Props) => {
+  const dataRef = useRef<(AggregatedLedgerEntriesEdge | null)[] | null>(null);
+  const totalCountRef = useRef<number>();
+  const newRows = useRef(0);
+
+  const variables = useMemo<LedgerEntriesQueryVariables>(
+    () => ({
+      partyId,
+      dateRange: filter?.vegaTime?.value,
+      pagination: { first: !filter?.vegaTime?.value ? 100 : undefined },
```

Comment:
suggestion (non blocking): I think it would make more sense to have a default filter applied, say a range of one week ago to present, rather than an arbitrary number. This is because you are essentially applying a filter to the user but it won't be represented in the UI anywhere

Reply:
I would rather leave the pagination filter unset for now. Even if we add a date range we would still be applying a filter. 
also in orders provider we are displaying first:100 and I am not sure what was the reasoning behind that. perhaps @bglownia knows more on how this works on reload